worlds/landstalker/Items.py

```diff
@@ -0,0 +1,107 @@
+    "Purple Jewel":         LandstalkerItemData(42, ItemClassification.progression, 500,    0),
+    "Gola's Eye":           LandstalkerItemData(43, ItemClassification.progression, 400,    1),
+    "Death Statue":         LandstalkerItemData(44, ItemClassification.filler,      150,    1),
+    "Dahl":                 LandstalkerItemData(45, ItemClassification.filler,      100,    18),
+    "Restoration":          LandstalkerItemData(46, ItemClassification.filler,      40,     9),
+    "Logs":                 LandstalkerItemData(47, ItemClassification.progression, 100,    2),
+    "Oracle Stone":         LandstalkerItemData(48, ItemClassification.progression, 250,    1),
+    "Idol Stone":           LandstalkerItemData(49, ItemClassification.progression, 200,    1),
+    "Key":                  LandstalkerItemData(50, ItemClassification.progression, 150,    1),
+    "Safety Pass":          LandstalkerItemData(51, ItemClassification.progression, 250,    1),
+    "Green Jewel":          LandstalkerItemData(52, ItemClassification.progression, 500,    0),  # No52 in base game
+    "Bell":                 LandstalkerItemData(53, ItemClassification.useful,      200,    1),
+    "Short Cake":           LandstalkerItemData(54, ItemClassification.useful,      250,    1),
+    "Gola's Nail":          LandstalkerItemData(55, ItemClassification.progression, 800,    1),
+    "Gola's Horn":          LandstalkerItemData(56, ItemClassification.progression, 800,    1),
+    "Gola's Fang":          LandstalkerItemData(57, ItemClassification.progression, 800,    1),
+    # Broad Sword: 58
+    # Leather Breast: 59
+    # Leather Boots: 60
+    # No Ring: 61
+    "Life Stock":           LandstalkerItemData(62, ItemClassification.filler,      250,    0),  # Variable amount
+    "No Item":              LandstalkerItemData(63, ItemClassification.filler,      0,      0),
+    "1 Gold":               LandstalkerItemData(64, ItemClassification.filler,      1,      1),
+    "20 Golds":             LandstalkerItemData(65, ItemClassification.filler,      20,     15),
+    "50 Golds":             LandstalkerItemData(66, ItemClassification.filler,      50,     7),
+    "100 Golds":            LandstalkerItemData(67, ItemClassification.filler,      100,    5),
+    "200 Golds":            LandstalkerItemData(68, ItemClassification.useful,      200,    2),
+
+    "Progressive Armor":    LandstalkerItemData(69, ItemClassification.useful,      250,    0)
+}
+
+
+def get_weighted_filler_item_names():
+    weighted_item_names: List[str] = []
+    for name, data in item_table.items():
+        if data.classification == ItemClassification.filler:
+            weighted_item_names += [name for _ in range(0, data.quantity)]
+    return weighted_item_names
+
+
+def build_item_name_to_id_table():
+    item_name_to_id_table = {}
+    for name, data in item_table.items():
+        item_name_to_id_table[name] = data.id + BASE_ITEM_ID
```

Comment:
Not sure I understand what would any method from the `Random` class accomplish here, since we are just building a simple matching table.

The way you do it in messenger is as follows :
```py
    item_name_to_id = {item: item_id for item_id, item in enumerate(ALL_ITEMS, base_offset)}
```